### ISSUE TITLE
"elided_named_lifetimes" warning (appears to be) unresolveable 🤔

### ISSUE THREAD

**Issue by mitchhentges** [rust-lang/rust#133982]
I tried this code ([playground here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4889bc6d242179748417b60443712b86)):

```rust
struct _Foo<'k, V> {
    pub _field: &'k V,
}

struct _Struct {}

impl _Struct {
    fn _at<'a>(&'a self, _param: &'a i32) -> _Foo<&'a i32> {
        unimplemented!();
    }
}

fn main() {}
```

I expected to see it `cargo build` without a warning.

Instead, I'm seeing the following warning:
```
warning: elided lifetime has a name
 --> src/main.rs:8:50
  |
8 |     fn _at<'a>(&'a self, _param: &'a i32) -> _Foo<&'a i32> {
  |            -- lifetime `'a` declared here        ^ this elided lifetime gets resolved as `'a`
  |
  = note: `#[warn(elided_named_lifetimes)]` on by default
```

### Meta

(stable) `rustc --version --verbose`:
```
rustc 1.83.0 (90b35a623 2024-11-26)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: x86_64-unknown-linux-gnu
release: 1.83.0
LLVM version: 19.1.1
```
(nightly) `rustc --version --verbose`:
```
rustc 1.85.0-nightly (c94848c04 2024-12-05)
binary: rustc
commit-hash: c94848c046d29f9a80c09aae758e27e418a289f2
commit-date: 2024-12-05
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.5
```

-----

Interestingly, if I remove the explicit lifetime from the return value (`-> _Foo<&i32>` instead), I get duplicate warnings:
```
warning: elided lifetime has a name
 --> src/main.rs:8:50
  |
8 |     fn _at<'a>(&'a self, _param: &'a i32) -> _Foo<&i32> {
  |            -- lifetime `'a` declared here        ^ this elided lifetime gets resolved as `'a`
  |
  = note: `#[warn(elided_named_lifetimes)]` on by default

warning: elided lifetime has a name
 --> src/main.rs:8:51
  |
8 |     fn _at<'a>(&'a self, _param: &'a i32) -> _Foo<&i32> {
  |            -- lifetime `'a` declared here         ^ this elided lifetime gets resolved as `'a`

warning: `splitto-kvitto` (bin "splitto-kvitto") generated 2 warnings
```

-----

I'm not sure if this is a bug or if I'm misunderstanding the warning and not seeing the solution :)
Thanks!

**Comment by mitchhentges** [comment]
Ah, I needed to do `-> _Foo<&'a, '&a i32>`!
